### ISSUE TITLE
dynamic import of async module from worker

### ISSUE THREAD

**Issue by DPOH-VAR** [denoland/deno#24069]
Version: Deno 1.44.0

Worker can not import async modules

`main.ts`:
```ts
new Worker(import.meta.resolve("./worker.ts"), {type: "module"});
```
`worker.ts`:
```ts
console.log("worker-start");
await import("./async.ts");
console.log("worker-end");
```
`async.ts`:
```ts
console.log('async-start');
await new Promise(r => setTimeout(r, 1000));
console.log('async-end');
```
---
run `deno run -A ./main.ts`
Output:
```
worker-start
async-start
async-end
```
Expected:
```
worker-start
async-start
async-end
worker-end
```

**Comment by dsherret** [comment]
@nathanwhit is this related to https://github.com/denoland/deno/issues/24098 ?

**Comment by nathanwhit** [comment]
> @nathanwhit is this related to #24098 ?

Yup! This will be fixed once #24128 lands.

**Comment by lucacasonato** [comment]
This should be fixed. @DPOH-VAR Can you try again in deno canary? (`deno upgrade --canary`)

**Comment by lucacasonato** [comment]
I confirmed this is fixed in `deno 1.44.1+d74be08`.
